pkg/k8s/wrapper.go

```diff
@@ -94,6 +96,15 @@ func NewK8sWrapper(client client.Client, coreV1 corev1.CoreV1Interface, ctx cont
 	return &k8sWrapper{cacheClient: client, eventRecorder: recorder, context: ctx}
 }
 
+func (k *k8sWrapper) GetDeployment(namespace string, name string) (*appV1.Deployment, error) {
+	deployment := &appV1.Deployment{}
+	err := k.cacheClient.Get(k.context, types.NamespacedName{
```

Comment:
it's a bad practice to store context objects. let's refactor this in future versions given it's already like this in other apis.